src/utils/MastodonUtils.C

```diff
@@ -68,3 +71,125 @@ MastodonUtils::regularize(const std::vector<Real> & history_acc,
   }
   return {reg_tme, reg_acc};
 }
+
+bool
+MastodonUtils::checkVectorsUnequalSize(std::vector<std::vector<Real>> & vectors)
+{
+  for (auto v : vectors)
+  {
+    if (v.size() != vectors[0].size())
+      return true;
+  }
+  return false;
+}
+
+bool
+MastodonUtils::areVectorsEqual(std::vector<Real> & vector1,
+                               std::vector<Real> & vector2,
+                               Real percent_error)
+{
+  if (vector1.size() != vector2.size())
+    return false;
+  for (std::size_t i = 0; i < vector1.size(); ++i)
+  {
+    if (!MooseUtils::absoluteFuzzyEqual(
+            vector1[i], vector2[i], std::abs(vector1[i] * percent_error / 100)))
+      return false;
+  }
+  return true;
+}
+
+bool
+MastodonUtils::isVectorNonPositive(const std::vector<Real> & vector)
+{
+  for (auto element : vector)
+    if (element <= 0)
+      return true;
+  return false;
+}
+
+Real
+MastodonUtils::mean(std::vector<Real> & vector)
+{
+  Real sum = std::accumulate(vector.begin(), vector.end(), 0.0);
+  return sum / vector.size();
+}
+
+Real
+MastodonUtils::median(std::vector<Real> & vector, const std::string & interpolation)
+{
+  std::sort(vector.begin(), vector.end());
+  if (vector.size() % 2 != 0)
+    return vector[(vector.size() - 1) / 2];
+  else if (interpolation == "linear")
+    return (vector[vector.size() / 2] + vector[vector.size() / 2 - 1]) / 2.0;
+  else if (interpolation == "lower")
+    return vector[vector.size() / 2 - 1];
+  else if (interpolation == "higher")
+    return vector[vector.size() / 2];
+  else
+    mooseError("Invalid interpolation type in median calculation.");
+}
+
+Real
+MastodonUtils::percentile(std::vector<Real> & vector,
```

Comment:
inputs should all be const

Reply:
Same here. Doing a sort on the vector, so not using const.

Reply:
Same as above for median.